text/entities/0264-resource-and-entities.md

```diff
@@ -0,0 +1,284 @@
+- Downstream components should be able to engage with the Entity model in Resource.
+
+The following changes are made:
+
+### Resource
+
+| Field | Type | Description | Changes |
+| ----- | ---- | ----------- | ------- |
+| schema_url | string | The Schema URL, if known. This is the identifier of the Schema that the resource data  is recorded in.  This field is deprecated and should no longer be used. | Will be deprecated |
+| dropped_attributes_count |  integer | dropped_attributes_count is the number of dropped attributes. If the value is 0, then no attributes were dropped. | Unchanged |
+| attributes | repeated KeyValue | Set of attributes that describe the resource.<br/><br/>Attribute keys MUST be unique (it is not allowed to have more than one attribute with the same key).| Unchanged |
+| entities | repeated ResourceEntityRef | Set of entities that participate in this Resource. | Added |
+
+The DataModel would ensure that attributes in Resource are produced from both the identifying and descriptive attributes of Entity.  This does not mean the protocol needs to transmit duplicate data, that design is TBD.
+
+### ResourceEntityRef
+
+The entityref data model, would have the following changes from the original [entity OTEP](https://github.com/open-telemetry/oteps/blob/main/text/entities/0256-entities-data-model.md) to denote references within Resource:
+
+| Field | Type | Description | Changes |
+| ----- | ---- | ----------- | ------- |
+| schema_url | string | The Schema URL, if known. This is the identifier of the Schema that the entity data  is recorded in. To learn more about Schema URL ([see docs](https://opentelemetry.io/docs/specs/otel/schemas/#schema-url)) | added |
+| type | string | Defines the type of the entity. MUST not change during the lifetime of the entity. For example: "service" or "host". This field is required and MUST not be empty for valid entities. | unchanged |
+| identifying_attributes_keys | repeated string | Attribute Keys that identify the entity.<br/>MUST not change during the lifetime of the entity. The Id must contain at least one attribute.<br/><br/>These keys MUST exists in Resource.attributes.<br/><br/>Follows OpenTelemetry common attribute definition. SHOULD follow OpenTelemetry semantic conventions for attributes.| now a reference |
+| descriptive_attributes_keys | repeated string | Descriptive (non-identifying) attribute keys of the entity.<br/>MAY change over the lifetime of the entity. MAY be empty. These attribute keys are not part of entity's identity.<br/><br/>These keys MUST exist in Resource.attributes.<br/><br/>Follows any value definition in the OpenTelemetry spec - it can be a scalar value, byte array, an array or map of values. Arbitrary deep nesting of values for arrays and maps is allowed.<br/><br/>SHOULD follow OpenTelemetry semantic conventions for attributes.| now a reference |
```

Comment:
Why do we need descriptive_attributes_keys in the the `EntityRef`? `EntityRef` _references_ an entity. It sufficient that a reference specifies the data that _identifies_ the entity. The `Type` and `Id` fields are sufficient for identifying the entity. Why do we need to specify descriptive attributes?

Reply:
We have descriptive attributes in Resource today, whether we like it or not.  This allows those use cases to continue, but we can directly address and deal with them.

Reply:
I think that's fine, we should keep descriptive attributes in the Resource. We don't have to have them in the EntityRefs though. Resource attributes will be a superset of EntityRef attributes.

Reply:
Then is the only difference between an entity and an entity ref within resources the lack/presence of the descriptive attributes?

Reply:
I think we need to keep descriptive attributes so we know how to handle certain entity-merge issues.  Please see use cases listed below.

The main differences between `Entity` and `EntityRef`;

- `EntityRef` is embedded on entity.
- `EntityRef` implies a relationship between all entities on the resource (without any details on what that relationship is, besides a possible bundle/aggregation slice)
- `Entity` would be reported via an independent channel (TBD: current thinking is through events on OTLP logs).